### DISCUSSION TITLE
Warning: numerical errors at iteration 0

### DISCUSSION THREAD

**Discussion by rayjyh** [saibalmars/GraphRicciCurvature#27] [General]
I was trying to use ORC to detect communities in a weighted undirected graph. I encountered two errors: 

- >No cutoff point found!
- > Message: 'Neighbor weight sum too small, list:'
Arguments: ([(0.0, 1), (0.0, 2), (0.0, 3), (0.0, 4), (0.0, 5), (0.0, 6), (0.0, 7), (0.0, 8), (0.0, 9), (0.0, 10), (0.0, 11), (0.0, 12), (0.0, 13), (0.0, 14), (0.0, 15), (0.0, 16), (0.0, 17),)
Traceback (most recent call last):
Message: 'Neighbor weight sum too small, list:'
Arguments: ([(0.0, 0), (0.0, 501), (0.0, 502), (0.0, 503), (0.0, 504), (0.0, 505), (0.0, 506), (0.0, 507), (0.0, 508), (0.0, 509), (0.0, 510), (0.0, 511), (0.0, 512), )
Process ForkPoolWorker-1:
Process ForkPoolWorker-6:
Process ForkPoolWorker-2:

Any help will be highly appreciated!

**Comment by saibalmars** [comment]
Hi, 

The initial edge weight in the Ricci flow process is less meaningful since with iteration long enough, any initial edge weight set should ideally converge to the same set of edge weights. So I suggest just ignore the given weight (set all edge weights as 1 in the beginning) and see what's the result. 

For some graphs that the community structure is weaker, the auto-detect algorithm might fail. In this case. I suggest to run the Ricci flow process for 20 iterations and open it in `gephi` (via .gexf file) to check the cut result. By algorithm limitation, the graph with less community structure won't perform that well.

Let me know if you have more questions :)
